protected/humhub/components/behaviors/PolymorphicRelation.php

```diff
@@ -169,14 +169,14 @@ private function validateUnderlyingObjectType(?object $object)
     /**
      * Loads an active record based on classname and primary key.
      *
-     * @param string $className
+     * @param string|mixed $className
      * @param string|int $primaryKey
      *
      * @return null|ActiveRecord|ActiveRecordInterface
      */
-    public static function loadActiveRecord(string $className, $primaryKey): ?ActiveRecordInterface
+    public static function loadActiveRecord($className, $primaryKey): ?ActiveRecordInterface
     {
-        if (empty($className) || empty($primaryKey)) {
+        if (($primaryKey !== 0 && empty($primaryKey)) || !is_string($className) || trim($className) === '') {
```

Comment:
I would actually prefer the previous variant here for reasons of readability.

Not sure we need to consider a PK with ID 0 or the string of spaces.  Or do you see a specific reason for this?

Reply:
`0` is a valid primary key. I know that in most places you use `empty()`-checks, but I think that should be avoided.

The `trim()` is not required, of course. Happy to remove.

Reply:
You are right, it is a valid PK, but normally there are no situations in which this PK is assigned. e.g. `auto_increment` starts with 1.

I want to keep the codebase as consistent and simple as possible - so I don't want to implement an extra check for `0` here. If it becomes necessary to allow 0, we need to do changes in more places.